addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -368,6 +373,10 @@ func (h *Helper) HandleExistingClusterBootstrap(clusterBootstrap *runtanzuv1alph
 			}
 		}
 
+		if _, ok := clusterBootstrap.Annotations[constants.UnmanagedCNI]; ok {
+			clusterBootstrapTemplate.Spec.CNI = clusterBootstrap.Spec.CNI
```

Comment:
why to we need to set the CBT.spec.CNI when the unmanagedCNI is used?

Reply:
in order to leave the CNI part untouched (unmanaged) it's easier to set the templateCB.Sepc.CNI  to the same value of the currentCB.Spec.CNI before making the call to AddMissingSpecFieldsFromTemplate.  
That way currentCB.Spec.CNI stays that same.  Otherwise we would have to heavily modify AddMissingSpecFieldsFromTemplate to handle the case where clusterBootstrap.Annotations[constants.UnmanagedCNI]

We could set templateCB.Spec.CNI = nil but since technically "unmanaged" means we don't change it, we leave the setting to nil to the default webhook, and take the full meaning of unmanaged, which means we don't change it at all.